internal/pkg/cli/deploy/static_site.go

```diff
@@ -87,20 +94,23 @@ func (d *staticSiteDeployer) DeployWorkload(in *DeployWorkloadInput) (ActionReco
 
 // UploadArtifacts uploads static assets to the app stackset bucket.
 func (d *staticSiteDeployer) UploadArtifacts() (*UploadArtifactsOutput, error) {
+	return d.uploadArtifacts(d.uploadStaticFiles, d.uploadArtifactsToS3, d.uploadCustomResources)
+}
+
+func (d *staticSiteDeployer) uploadStaticFiles(out *UploadArtifactsOutput) error {
 	for _, f := range d.staticSiteMft.FileUploads {
-		if _, err := d.uploadFn(d.fs, filepath.Join(f.Context, f.Source), f.Destination,
-			&asset.UploadOpts{
-				Reincludes: f.Reinclude.ToStringSlice(),
-				Excludes:   f.Exclude.ToStringSlice(),
-				Recursive:  f.Recursive,
-				UploadFn: func(key string, contents io.Reader) (string, error) {
-					return d.s3Client.Upload(d.bucketName, key, contents)
-				},
-			}); err != nil {
-			return nil, err
+		assets, err := d.uploader.UploadToCache(filepath.Join(f.Context, f.Source), f.Destination, &asset.UploadOpts{
+			Reincludes: f.Reinclude.ToStringSlice(),
+			Excludes:   f.Exclude.ToStringSlice(),
+			Recursive:  f.Recursive,
+		})
+		if err != nil {
+			return err
```

Comment:
Do we need to wrap this error?

Reply:
i think the most useful context is in the errors already - errors are:
- "walk the file tree rooted at <source directory from `f`>"
- "upload <specific file path>"

we _could_ prefix those errors with the index of the upload if you think that's nice? something like:
"upload #1/4: upload <specific file path>: some error" 

thoughts?

Reply:
yeah that sounds good to me

Reply:
sg, i'll make the change!